rpc_util_test.go

```diff
@@ -290,3 +293,138 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
+	}{
+		{
+			name:       "Decompresses successfully with sufficient buffer size",
+			compressor: c,
+			input: func() mem.BufferSlice {
+				compressedData := compressData([]byte("decompressed data"))
+				return mem.BufferSlice{mem.NewBuffer(&compressedData, nil)}
+			}(),
+			maxReceiveMessageSize: 100,
+			want: func() mem.BufferSlice {
+				decompressed := []byte("decompressed data")
+				return mem.BufferSlice{mem.NewBuffer(&decompressed, nil)}
+			}(),
+			compressedSize: len([]byte("decompressed data")),
+			error:          nil,
+		},
+		{
+			name:                  "Fails with decompression error on empty input",
+			compressor:            c,
+			input:                 mem.BufferSlice{},
+			maxReceiveMessageSize: 1,
+			want:                  nil,
+			compressedSize:        0,
+			error:                 errors.New("decompression error"),
+		},
+		{
+			name:       "Fails with overflow error when decompressed size exceeds maxReceiveMessageSize",
+			compressor: c,
+			input: func() mem.BufferSlice {
+				compressedData := compressData([]byte("overflow test data"))
+				return mem.BufferSlice{mem.NewBuffer(&compressedData, nil)}
+			}(),
+			maxReceiveMessageSize: 5,
+			want:                  nil,
+			compressedSize:        6, //len([]byte("overflow test data")),
+			error:                 errors.New("overflow: received message size is larger than the allowed maxReceiveMessageSize (5 bytes)."),
+		},
+		{
+			name:       "Returns EOF error when maxReceiveMessageSize is MaxInt64 but data ends prematurely",
+			compressor: c,
+			input: func() mem.BufferSlice {
+				compressedData := compressData([]byte("small data"))
+				return mem.BufferSlice{mem.NewBuffer(&compressedData, nil)}
+			}(),
+			maxReceiveMessageSize: math.MaxInt64,
+			want: func() mem.BufferSlice {
+				smallDecompressed := []byte("small data")
+				return mem.BufferSlice{mem.NewBuffer(&smallDecompressed, nil)}
+			}(),
+			compressedSize: len([]byte("small data")),
+			error:          nil,
+		},
+		{
+			name:                  "Fails with io.Copy error during decompression",
+			compressor:            c,
+			input:                 mem.BufferSlice{},
+			maxReceiveMessageSize: 100,
+			want:                  nil,
+			compressedSize:        0,
+			error:                 errors.New("simulated io.Copy read error"),
+		},
+		{
+			name:       "Succeeds when message size is much smaller than maxReceiveMessageSize",
+			compressor: c,
+			input: func() mem.BufferSlice {
+				compressedData := compressData([]byte("small message"))
+				return mem.BufferSlice{mem.NewBuffer(&compressedData, nil)}
+			}(),
+			maxReceiveMessageSize: 50,
+			want: func() mem.BufferSlice {
+				decompressed := []byte("small message")
+				return mem.BufferSlice{mem.NewBuffer(&decompressed, nil)}
+			}(),
+			compressedSize: len([]byte("small message")),
+			error:          nil,
+		},
+		{
+			name:       "Succeeds with MaxInt64 receive size and small decompressed data",
```

Comment:
this is redundant as well. Case 3) should already test it. I think its hard to make a unit test that can exceed MaxInt64 threshold

Reply:
done